x/concentrated-liquidity/fees_test.go

```diff
@@ -651,7 +670,12 @@ func (s *KeeperTestSuite) TestCollectFees() {
 			clKeeper := s.App.ConcentratedLiquidityKeeper
 			ctx := s.Ctx
 
-			s.initializeFeeAccumulatorPositionWithLiquidity(ctx, validPoolId, tc.owner, tc.lowerTick, tc.upperTick, tc.initialLiquidity)
+			// Position needs to be set in the store before calling CollectFees.
+			// This is because we iterate through all positions instead of having to call each position with separate frozenUntil times.
+			position := model.Position{Liquidity: tc.initialLiquidity, FrozenUntil: tc.frozenUntil}
+			s.App.ConcentratedLiquidityKeeper.SetPosition(ctx, validPoolId, ownerWithValidPosition, tc.lowerTick, tc.upperTick, &position, tc.frozenUntil)
```

Comment:
Note to reviewer: Without setting the position, we don't call initializePosition and therefore we wont have any positions in store. Without any positions in store, when we go to call `CollectFees` it wont have any positions to iterate over. 

There are no problems here, just wanted to explain why setting the position was suddenly necessary (now that we iterate over positions from the store when collecting fees)

Reply:
Ditto for the question above, why do we need to set position in the first place when we have the has position check from collectFees?

Reply:
See this comment https://github.com/osmosis-labs/osmosis/pull/4109#discussion_r1089160947

Good catch :)